server/src/main/scala/org/allenai/scienceparse/FeedbackStore.scala

```diff
@@ -112,20 +114,31 @@ object FeedbackStore extends Logging {
     }
   }
 
-  def getAllFeedback: Traversable[(String, LabeledData)] = {
+  /**
+    * @param onOrAfter If given, constrains returned feedback to those added on or after this timestamp.
+    * @param before If given, constrains returned feedback to those added before this timestamp.
+    */
+  def getAllFeedback(
+    onOrAfter: Option[Instant] = None,
+    before: Option[Instant] = None
+  ): Traversable[(String, LabeledData)] = {
     import spray.json._
     import LabeledDataJsonProtocol._
 
+    val onOrAfterClause = onOrAfter.map(ts => sqls" AND a.timeadded >= $ts").getOrElse(sqls"")
+    val beforeClause = before.map(ts => sqls" AND a.timeadded < $ts").getOrElse(sqls"")
```

Comment:
`sqls""`? What is this sorcery? Is it just for this case?

Reply:
Yeah `sqls` is for sql fragments. Because this query is implemented with the interpolated `sql` thing, you can't use regular strings anywhere. I guess it's good that this sqls thing exists because I would have had to recode this whole method without it.

Reply:
🦈